### PR TITLE
[core] Spill at least the object fusion size instead of at most

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

Copied from #22571:

Whenever we spill, we try to spill all spillable objects. We also try to fuse small objects together to reduce total IOPS. If there aren't enough objects in the object store to meet the fusion threshold, we spill the objects anyway to avoid liveness issues. However, currently we spill at *most* the object fusion size when instead we should be spilling at *least* the fusion size. Then we use the max number of fused objects as a cap.

This PR fixes the fusion behavior so that we always spill at minimum the fusion size. If we reach the end of the spillable objects, and we are under the fusion threshold, we'll only spill it if we don't have other spills pending too. This gives the pending spills time to finish, and then we can re-evaluate whether it's necessary to spill the remaining objects. Liveness is also preserved.

Increases some test timeouts to allow tests to pass.

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
Didn't review the PR, but I assume the bug was fixed..

**Comment by kfstorm** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've updated our formatting configuration for C++ code. (see #22725)

This PR includes C++ code change. To prevent issues with merging your code, here's what you'll need to do:

1. Merge the latest changes from upstream/master branch into your branch.

```bash
git pull upstream master
git merge upstream/master
```

2. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated C++ formatting configuration.

3. Format changed files.

```bash
scripts/format.sh
```

4. Commit your changes.

```bash
git add --all
git commit -m "Format C++ code"
```

**Comment by stephanie-wang** [comment]
@rkooo567 I updated this PR a bit to address another issue I found where we're spilling at most the fusion size instead of at least. Can you take a look at the C++ changes and re-approve?

**Review by rkooo567 — CHANGES_REQUESTED** [review]
Temporarily request changed when I review the new change (will do this sun)

File: src/ray/raylet/local_object_manager.cc [hunk 1]
```diff
@@ -181,8 +181,7 @@ bool LocalObjectManager::SpillObjectsOfSize(int64_t num_bytes_to_spill) {
   auto it = pinned_objects_.begin();
   std::vector<ObjectID> objects_to_spill;
   int64_t counts = 0;
-  while (bytes_to_spill <= num_bytes_to_spill && it != pinned_objects_.end() &&
-         counts < max_fused_object_count_) {
+  while (it != pinned_objects_.end() && counts < max_fused_object_count_) {
```
Comment (rkooo567) [inline src/ray/raylet/local_object_manager.cc:184]: Can you update the docstring of `SpillObjectsOfSize` to explain semantic changes?

**Review by rkooo567 — APPROVED** [review]
I think we should update the docstring, but other than that, it lgtm

**Pull request merged**